frontend/src/store/reducers/UserReducer/ActionCreator.ts

```diff
@@ -1,28 +1,25 @@
-import { createAsyncThunk, SerializedError } from "@reduxjs/toolkit";
-import { ErrorResponse } from "@rtk-query/graphql-request-base-query/dist/GraphqlBaseQueryTypes";
-import { Exact, LoginMutation, UsersPermissionsLoginInput } from "generated/graphql";
+import { createAction, createAsyncThunk } from "@reduxjs/toolkit";
+import api from "api/axios";
+import { BASE_URL } from "config";
+import { IUser } from "types/User";
 
-interface LoginUserCreatorData {
-  loginMutationTrigger: (arg: Exact<{data: UsersPermissionsLoginInput;}>) =>
-    Promise<{data: LoginMutation;} | {error: ErrorResponse | SerializedError;}>,
-  credentials: LoginUserCredentials, 
-}
-interface LoginUserCredentials {
-  email: string,
-  password: string,
-}
+export const addUser = createAction<IUser>('add/user');
 
-export const loginUser = createAsyncThunk(
-  'user/login',
-  async (creatorData: LoginUserCreatorData, thunkAPI) => {
-    const {loginMutationTrigger, credentials} = creatorData;
-    const response = await loginMutationTrigger({data: {identifier: credentials.email, password: credentials.password}});
-    
-    if ((response as {error: ErrorResponse | SerializedError;}).error) {
-      return thunkAPI.rejectWithValue('Something went wrong.');
-    }
+export const fetchUser = createAsyncThunk(
```

Comment:
If you were to use redux-saga you might like this:

import axios, { AxiosRequestConfig, AxiosResponse, AxiosInstance } from 'axios';
import { put, call, select, CallEffect, PutEffect, SelectEffect } from 'redux-saga/effects';
import { validateStatus } from '../../utils/validateStatus';
import { disconnectWalletState } from '../user/reducer';
import userSelector from '../user/selectors';

const client: AxiosInstance = axios.create({
  baseURL: 'https://dev2-totoro.rocknblock.io/api/v1/',
  headers: {
    'Access-Control-Allow-Origin': '*',
    // 'Content-Type': 'application/json',
  },
  validateStatus,
});

export default function* ajax(
  config: AxiosRequestConfig,
): Generator<SelectEffect | CallEffect | PutEffect> {
  const accessToken = yield select(userSelector.getProp('key'));

  if (accessToken) {
    client.defaults.headers.common['Authorization'] = `Bearer ${accessToken}`;
  }

  // @ts-ignore
  const response: AxiosResponse<ApiResponse<unknown>> = yield call<(configVar: AxiosRequestConfig) => void
  >(client, config);

  if (accessToken && response.status === 401) {
    yield put(disconnectWalletState());
  }

  return response;
}